src/doc/src/reference/unstable.md

```diff
@@ -1929,6 +1930,31 @@ For example:
 cargo +nightly fix -Zfix-edition=end=2024,future
 ```
 
+## check-target-cfgs
+
+* Tracking Issue: [#00000](https://github.com/rust-lang/cargo/issues/00000)
+
+This feature checks for unexpected cfgs in `[target.'cfg(...)']` entries, based
+on `rustc --print=check-cfg`.
+
+```sh
+cargo check -Zcheck-target-cfgs
+```
+
+It follows the lint Rust `unexpected_cfgs` lint configuration:
+
+```toml
+[target.'cfg(foo)'.dependencies]
+cfg-if = "1.0"
+
+[lints.cargo.unexpected_cfgs]
+level = "deny"
+
+[lints.rust.unexpected_cfgs]
+level = "warn"
+check-cfg = ['cfg(foo)']
```

Comment:
> I don't remember if we had settle on the lint config position, ie if we should share check-cfg = [...] from rust.unexpected_cfgs or if cargo.unexpected_cfgs should have a distinct one. I've left the sharing for now.

I would say if we have our own lint, we have our own lint config.

If anything, in light of lints being fractured like this, I would see us re-evaluating whether we should have per-lint config or a general config table.